main.tf

```diff
@@ -584,6 +595,12 @@ resource "azurerm_kubernetes_cluster" "main" {
   }
 }
 
+resource "null_resource" "aks_cluster_recreate" {
```

Comment:
I don't understand this name ? Is it attempting to recreate the all cluster if the noproxy value changes ?

I dont see the corresponding `azapi_update_resource` with the `lifecycle.replace_triggered_by` section. Is it missing ?

Reply:
You're correct, I found that we can update this property instead of create the whole cluster. I'll update this pr.